### ISSUE TITLE
Feature suggestion: rules with different properties for specific files

### ISSUE THREAD

**Issue by webimpress** [squizlabs/PHP_CodeSniffer#2258]
I have a feature suggestion to add ability to define different rule properties for specific files, let say something like:

```xml
<rule ref="MyRule.Ref">
  <include-pattern>./src</include-pattern>
  <properties>
    <property name="indent" value="4">
  </properties>
</rule>

<rule ref="MyRule.Ref"> <!-- the same rule
  <include pattern>./other</include-pattern> <!-- different files
  <properties>
    <property name="indent" value="2"> <!-- here different value
  </properties>
</rule>
```

Not sure what should be the format, maybe we can do it another way... Or maybe it is already possible to do something like that, but I am not aware of it?

What do you think? If you like that idea maybe I would be able to work on it in some near future ;-)

**Comment by jrfnl** [comment]
I like this idea.

Format-wise, I think we'd need a way to still allow a property to be set globally for a rule, while also allowing file-specific deviations. Maybe something along the lines of the below ?

```xml
<file>./src</file>

<rule ref="MyRule.Ref">
  <properties>
    <property name="indent" value="4">
    <property name="indent" value="2" include-pattern="./other">
  </properties>
</rule>
```

**Comment by jrfnl** [comment]
Loosely related to #2126
